### PR TITLE
fix: icx-proxy should only consult query string and referrer when run locally

### PR DESCRIPTION
**Pull request by ericswanson-dfinity**
Only check the URI (query string) and `referer` header when `icx-proxy` is run for local development, determined by a new `--local` command-line argument.

### CODE REVIEW

**Comment by nomeata** [comment]
Thanks!

File: icx-proxy/src/main.rs [hunk 1]
```diff
@@ -465,7 +461,9 @@ fn main() -> Result<(), Box<dyn Error>> {
 
     let counter = AtomicUsize::new(0);
     let debug = opts.debug;
-    let check_uri_and_referer_for_canister_id = opts.check_uri_and_referer_for_canister_id;
+
+    /// For local/development, check the URI (query string) and the referer for a canister id.
+    let check_uri_and_referer_for_canister_id = opts.address.ip().is_loopback();
```
Comment (nomeata) [inline icx-proxy/src/main.rs]: Nice in principle. But does that work? Are we sure that the deployed icx-proxy doesn’t sit behind some form of proxy _on the same machine_?

Comment (ericswanson-dfinity) [reply icx-proxy/src/main.rs]: Yeah, it does not work, both because of that and because in production icx-proxy binds to localhost, behind nginx

Comment (nomeata) [reply icx-proxy/src/main.rs]: Yes, that’s what what I mean. Too bad.

Comment (ericswanson-dfinity) [reply icx-proxy/src/main.rs]: Hh yes that is exactly what you wrote

**Comment by ericswanson-dfinity** [comment]
icx-proxy moved to https://github.com/dfinity/icx-proxy

**Pull request closed without merging**